### DISCUSSION TITLE
Index search and bookmark

### DISCUSSION THREAD

**Discussion by tacman** [meilisearch/product#785] [Feedback & Feature Proposal]
I have dozens of indexes, and am about to have thousands.  i can manage the indexes myself, but it'd be great if there were an api call that allowed me to search by name/pattern, number of documents, most recent change, etc.

On a related note, one of my biggest issues with the mini-dashboard is that I can't link directly to an index.  Instead, I have to select from a dropdown, and then I think it sets a cookie.  Better would be a URL that would open the index page.

**Comment by macraig** [comment]
Hi @tacman , can you expand on what you mean by managing the indexes? What would an example scenario look like in your proposed solution?

**Comment by tacman** [comment]
![image](https://github.com/user-attachments/assets/40e623fd-9749-4312-adeb-894cc3d119f7)

If I have 1000 indexes, but I only want the ones that begin with a prefix or end with a suffix, there's no way to do that now.  I can load all of them, then use javascript to filter, but for now there's no way to filter the indexes by a search pattern, only be listing them.

![image](https://github.com/user-attachments/assets/8c4ba706-a765-4e40-a67f-41457320296e)

So I have my own database of museums, each museum has different tables that are indexed (the objects, of course, but also collections, people, cultures, materials, epochs, images, organizations, etc.)  So I know I'm working with the walter museum and want to search their cultures table.  How can I link to the mini-dashboard?  I can't, I have to go to the home page, and select from my index list.

So by managing I mean that I'd like to have search features for the indexes themselves.  If there were only a search engine available....;-)

How meta -- managing the search index by creating an internal index where each index is a document.  This isn't a crazy idea.  In fact, what I'd _really_ like to do is tag the index, so I can filter them when searching.  In my case, I want to group by museum, of course, but also all the culture indexes, people indexes, etc.   Eventually I will be using this information for a federated search, so having the ability so say "do a federated search of all the indexes tagged as published and culture".

btw, that screen capture above comes from https://github.com/riccox/meilisearch-ui/, and the ability to filter was just added 2 days ago. https://github.com/riccox/meilisearch-ui/issues/134  It's a great dashboard for meili, the ability to do a full search from the dashboard, plus the ability to see tasks, are wonderful.  Check it out at https://meilisearch-ui.riccox.com/

**Reply by tacman** [reply]
And of course, I would like to sort the indexes by things like number of documents, most recently updated, etc.

**Reply by macraig** [reply]
Thanks for the detailed reply! I see how searching over indexes would be helpful to you and users who have a large number of indexes in general. We haven't planned to prioritize it yet, but we'll keep it in mind for our future planning sessions.

**Reply by tacman** [reply]
> If there were only a search engine available....;-)

did you laugh at this?  That was the intent.

**Reply by macraig** [reply]
I did, the irony is high on this one 😄
